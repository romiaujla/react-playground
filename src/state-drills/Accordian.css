body{
    background: snow;    
}

li{
    list-style: none;
    margin-bottom: 30px;
}

button{
    cursor: pointer;
    padding: 10px 20px;
    border: none;
    background: lightblue;
    text-transform: uppercase;
    font-weight: bold;
}

button:hover,
button:focus{
    background: cyan;
}

