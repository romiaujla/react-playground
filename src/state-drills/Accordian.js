import React from 'react';
import './Accordian.css';

class Accordian extends React.Component {

    static defaultProps = {
        sections: [
            {
                title: '',
                content: ''
            }
        ]
    }

    constructor(props){
        super(props);
        this.state = {
            currentAccordian: null
        }
    }

    // Handling the button click for changing the state of current accordian
    handleButtonClick = (index) => {
        this.setState({
            currentAccordian: index
        })
    }

    // renders the buttons
    renderButtons = (section, index) => {
        return (
            <button type='button' onClick={() => this.handleButtonClick(index)} >
                {section.title}
            </button>
        );
    }

    // Rendering the Accordian List
    renderAccordianList = () => {
        return this.props.sections.map((section, index) => (
            <li key={index} className='accordian-item'>
                {!!(section.title !== '') && this.renderButtons(section, index)}
                {!!(this.state.currentAccordian === index) && (<p>{section.content}</p>)}
            </li>
        ));
    }


    // Main render method
    render(){

        return (
            <ul className='accordian-component'>
                {this.renderAccordianList()}
            </ul>
        );
    }
}

export default Accordian;