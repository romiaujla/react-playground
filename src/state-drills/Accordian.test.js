import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordian';

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

describe('Accordian Component', () => {

    it('renders without crashing', ()=> {
        const div = document.createElement('div');
        ReactDOM.render(<Accordian />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders no sections by default', () =>{
        const wrapper = shallow(<Accordian sections={sections} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders an empty li with no props provided', () => {
        const wrapper = shallow(<Accordian />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it('renders the correct section on button click', () => {
        const wrapper = shallow(<Accordian sections={sections} />)
        wrapper.find('button').at(1).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('renders last section when multiple button clicks take place', () => {
        const wrapper = shallow(<Accordian sections={sections} />)
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
    })

});