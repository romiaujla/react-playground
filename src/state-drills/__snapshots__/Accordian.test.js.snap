// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component renders an empty li with no props provided 1`] = `
<ul
  className="accordian-component"
>
  <li
    className="accordian-item"
    key="0"
  />
</ul>
`;

exports[`Accordian Component renders last section when multiple button clicks take place 1`] = `
<ul
  className="accordian-component"
>
  <li
    className="accordian-item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="accordian-item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="accordian-item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordian Component renders no sections by default 1`] = `
<ul
  className="accordian-component"
>
  <li
    className="accordian-item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="accordian-item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="accordian-item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component renders the correct section on button click 1`] = `
<ul
  className="accordian-component"
>
  <li
    className="accordian-item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="accordian-item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="accordian-item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;
